engine/consensus/matching/engine_test.go

```diff
@@ -57,421 +52,187 @@ func TestMatchingEngine(t *testing.T) {
-				return nil
-			}
-			return block.Payload.Index()
-		},
-		func(blockID flow.Identifier) error {
-			block, found := ms.blocks[blockID]
-			if !found {
-				return storerr.ErrNotFound
-			}
-			if block.Payload == nil {
-				return storerr.ErrNotFound
-			}
-			return nil
-		},
-	)
-
-	// ~~~~~~~~~~~~~~~~ SETUP INCORPORATED RESULTS MEMPOOL ~~~~~~~~~~~~~~~~~ //
-	ms.pendingResults = make(map[flow.Identifier]*flow.IncorporatedResult)
-	ms.resultsPL = &mempool.IncorporatedResults{}
-	ms.resultsPL.On("Size").Return(uint(0)).Maybe() // only for metrics
-	ms.resultsPL.On("All").Return(
-		func() []*flow.IncorporatedResult {
-			results := make([]*flow.IncorporatedResult, 0, len(ms.pendingResults))
-			for _, result := range ms.pendingResults {
-				results = append(results, result)
-			}
-			return results
-		},
-	).Maybe()
-
-	// ~~~~~~~~~~~~~~~~~~~~~~ SETUP APPROVALS MEMPOOL ~~~~~~~~~~~~~~~~~~~~~~ //
-	ms.approvalsPL = &mempool.Approvals{}
-	ms.approvalsPL.On("Size").Return(uint(0)).Maybe() // only for metrics
-
-	// ~~~~~~~~~~~~~~~~~~~~~~~~ SETUP SEALS MEMPOOL ~~~~~~~~~~~~~~~~~~~~~~~~ //
-	ms.pendingSeals = make(map[flow.Identifier]*flow.IncorporatedResultSeal)
-	ms.sealsPL = &mempool.IncorporatedResultSeals{}
-	ms.sealsPL.On("Size").Return(uint(0)).Maybe() // only for metrics
-	ms.sealsPL.On("ByID", mock.Anything).Return(
-		func(sealID flow.Identifier) *flow.IncorporatedResultSeal {
-			return ms.pendingSeals[sealID]
-		},
-		func(sealID flow.Identifier) bool {
-			_, found := ms.pendingSeals[sealID]
-			return found
-		},
-	)
-
 	// ~~~~~~~~~~~~~~~~~~~~~~~ SETUP MATCHING ENGINE ~~~~~~~~~~~~~~~~~~~~~~~ //
 	ms.requester = new(module.Requester)
-	ms.assigner = &module.ChunkAssigner{}
+	ms.receiptValidator = &module.ReceiptValidator{}
 
 	ms.matching = &Engine{
 		unit:                    unit,
 		log:                     log,
 		engineMetrics:           metrics,
 		mempool:                 metrics,
 		metrics:                 metrics,
-		state:                   ms.state,
+		state:                   ms.State,
 		requester:               ms.requester,
-		resultsDB:               ms.resultsDB,
-		headersDB:               ms.headersDB,
-		indexDB:                 ms.indexDB,
-		incorporatedResults:     ms.resultsPL,
-		approvals:               ms.approvalsPL,
-		seals:                   ms.sealsPL,
+		resultsDB:               ms.ResultsDB,
+		headersDB:               ms.HeadersDB,
+		indexDB:                 ms.IndexDB,
+		incorporatedResults:     ms.ResultsPL,
+		receipts:                ms.ReceiptsPL,
+		approvals:               ms.ApprovalsPL,
+		seals:                   ms.SealsPL,
 		checkingSealing:         atomic.NewBool(false),
 		requestReceiptThreshold: 10,
 		maxResultsToRequest:     200,
-		assigner:                ms.assigner,
+		assigner:                ms.Assigner,
 		requireApprovals:        true,
+		receiptValidator:        ms.receiptValidator,
 	}
 }
 
-func (ms *MatchingSuite) TestOnReceiptInvalidOrigin() {
```

Comment:
checking Receipt validity is now handled by `ReceiptValidator`. For the matching engine, we only need to test that the output of `ReceiptValidator` is processed properly: see test https://github.com/onflow/flow-go/blob/32081ff9f91d24fd5c10ab57bdf447a70c7dd4ef/engine/consensus/matching/engine_test.go#L212-L213